src/components/user-list/user-list.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+import { Search } from '../search';
+import { User } from '../user';
+
+export class UserList extends Component {
+
+  componentDidMount() {
+    this.props.fetchUsers()
+  }
+  searchHandler = (event) => {
+    let { nativeEvent: { target: { value } } } = event;
+    this.props.searchInput(value);
+    this.props.filterUser();
```

Comment:
If you're trying to filter based on each user input, then this method shouldn't be called here, it should be called inside the saga against `searchInput`